myokit/formats/diffsl/_exporter.py

```diff
@@ -0,0 +1,523 @@
+                var.set_rhs(myokit.Multiply(var.rhs(), time_factor_inv))
+            for var in alphas:
+                var.set_rhs(myokit.Multiply(var.rhs(), time_factor_inv))
+            for var in betas:
+                var.set_rhs(myokit.Multiply(var.rhs(), time_factor_inv))
+            for var in taus:
+                var.set_rhs(myokit.Multiply(var.rhs(), time_factor))
+
+        # Add intermediary variables on rhs of state derivatives
+        self._prep_derivatives(model)
+
+        return model, currents
+
+    def _generate_diffsl(self, model, extra_out_vars=None):
+        """
+        Generate a DiffSL model from a prepped Myokit model.
+        DiffSL outputs will be set to state variables and extra_out_vars
+        """
+
+        # Create DiffSL-compatible variable names
+        var_to_name = self._create_diffsl_variable_names(model)
+
+        # Create a naming function
+        def var_name(e):
+            if isinstance(e, myokit.LhsExpression):
+                return var_to_name[e.var()]
+            elif isinstance(e, myokit.Variable):
+                return var_to_name[e]
+            raise ValueError(  # pragma: no cover
+                'Not a variable or LhsExpression: ' + str(e)
+            )
+
+        # Create an expression writer
+        e = diffsl.DiffSLExpressionWriter()
+        e.set_lhs_function(var_name)
+
+        export_lines = []  # DiffSL export lines
+        tab = '  '  # Tab character
+
+        # Sort equations in solvable order (grouped by component)
+        sorted_eqs = model.solvable_order()
+
+        # Variables to be excluded from output or handled separately.
+        # Derivatives and their intermediary variables (i.e. dot(x) and dot_x)
+        # are handled in dudt_i, F_i and G_i; time is excluded from the output;
+        # pace is made Vc.
+        time = model.time()
+        pace = model.binding('pace')
+        special_vars = set(
+            [time]
+            + [v.rhs().var() for v in model.states()]
+            + [v.lhs().var() for v in model.states()]
+        )
+        if pace is not None:
+            special_vars.add(pace)
+
+        # Add metadata
+        export_lines.append('/*')
+        export_lines.append('This file was generated by Myokit.')
+        if model.meta:
+            export_lines.append('')
+            for key, val in sorted(model.meta.items()):
+                export_lines.append(f'{key}: {val}')
+            export_lines.append('')
+        export_lines.append('*/')
+        export_lines.append('')
+
+        # Add empty input parameter list
+        export_lines.append('/* Input parameters */')
+        export_lines.append('in = [ ]')
+        export_lines.append('')
+
+        # Add placeholder protocol
+        if pace is not None:
+            export_lines.append('/* Voltage protocol [mV] */')
+            export_lines.append(
+                'Vc { -80 + 120 * heaviside(t-500) - 80 * heaviside(t-1000) }'
```

Comment:
Why this hardcoded voltage clamp protocol?

Reply:
That's a placeholder that makes it clear where to put the protocol and also provides an example. Should I remove it? Can probably replace it with a comment.
@MichaelClerx
@chonlei

Reply:
Made example protocol a comment in 1a21249

Reply:
Does it make sense to add extra documentation on creating protocols with ramps etc.?

Reply:
Leaving it in as an example makes sense to me. Not sure you want to add lots more though!